eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/ZmqWatcher.scala

```diff
@@ -100,9 +100,28 @@ class ZmqWatcher(chainHash: ByteVector32, blockCount: AtomicLong, client: Extend
           blockCount.set(count)
           context.system.eventStream.publish(CurrentBlockCount(count))
       }
-      client.rpcClient.invoke("getbalance").collect {
-        // We track our balance in mBTC because a rounding issue in BTC would be too impactful.
-        case JDecimal(balance) => Metrics.BitcoinBalance.withoutTags().update(balance.doubleValue * 1000)
+      // TODO: move to a separate actor that listens to CurrentBlockCount
+      client.rpcClient.invoke("listunspent").collect {
+        case JArray(values) =>
+          val utxos = values.map(utxo => {
+            val JInt(confirmations) = utxo \ "confirmations"
+            val JBool(safe) = utxo \ "safe"
+            val JDecimal(amount) = utxo \ "amount"
+            (amount.doubleValue * 1000, confirmations.toLong, safe)
+          })
+          val filteredByStatus = Seq(
+            (Monitoring.Tags.UtxoStatuses.Confirmed, utxos.filter(_._2 > 0)),
+            (Monitoring.Tags.UtxoStatuses.Unconfirmed, utxos.filter(_._2 == 0)),
+            (Monitoring.Tags.UtxoStatuses.Safe, utxos.filter(_._3)),
+            (Monitoring.Tags.UtxoStatuses.Unsafe, utxos.filter(!_._3)),
+          )
+          filteredByStatus.foreach {
+            case (status, filteredUtxos) =>
+              val amount = filteredUtxos.map(_._1).sum
+              log.info(s"we have ${filteredUtxos.length} $status utxos ($amount mBTC)")
```

Comment:
Would this read better if we printed one log line that has a mapping of status -> amount? A bit of log duplication (easy to misread) since we would have 4 such lines per block and potentially more if we ever make the statuses more granular

Reply:
IIUC, you'd like to replace the current multiple lines with one single log line that would look like:

```
unconfirmed -> 0.3 mBTC (7 utxos), confirmed -> 0.2 mBTC (3 utxo), safe -> 0.5 mBTC (5 utxo)
```

I prefer separating the log lines, it makes it easier to grep only the status you're interested in and automate analysis.

Reply:
sounds good

Reply:
This only counts the tips right?

Here is a script (courtesy @sstone) which also counts the number of unconfirmed txes that we can create:

```shell
#!/bin/bash

echo "utxos that can be created from confirmed utxos:"
echo `bitcoin-cli listunspent | jq '. | length * 25'`
echo "utxos that can be created from unconfirmed utxos:"
for i in `bitcoin-cli listunspent 0 | jq --raw-output '.[] | select(.confirmations == 0) | .txid'`
do
        bitcoin-cli getmempoolentry $i | jq '. | select(.ancestorcount == 1) | 25 - .descendantcount'
done | paste -sd+ | bc
```

Reply:
Is that really helpful to track as a metric?
We'd need to refresh that every time a new mempool entry consumes one of our unspent utxos, don't we?

The question is how do you want to use this metric.
The way I see it, it's very simple: if it's becoming low we create new utxos from our cold wallet (or something similar).
I don't think the added complexity is necessary in that case, is it?

Reply:
It's pretty useful, because this actually tracks our ability to publish unconfirmed txes, which the other values are a proxy for.

Reply:
But you agree that it's a lot harder to implement correctly, and more expensive as well (we'll need to recompute it basically every time the mempool changes if we want it to be correct).

Or maybe we don't care about such variations since we won't be able to react immediately anyway, and we're ok with computing this only once every block, regardless of the txs we'll submit to the mempool?

Reply:
Oh yes definitely, only once every block, not in realtime.

Reply:
Allright, I'll experiment with that.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1609/commits/e3d8f222254f704653682598399303a6a4d86b3a
I chose a slightly different route, because as discussed offline estimating the number of transactions we may be able to create can be wrong by an order of magnitude.
We instead simply discard unconfirmed utxos that have too many ancestors.
And when the metric is low, we should send more funds from a cold wallet.